patches/server/1045-More-Chest-Block-API.patch

```diff
@@ -0,0 +1,50 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: SoSeDiK <mrsosedik@gmail.com>
+Date: Wed, 1 May 2024 08:22:13 +0300
+Subject: [PATCH] More Chest Block API
+
+== AT ==
+public net.minecraft.world.level.block.ChestBlock isBlockedChestByBlock(Lnet/minecraft/world/level/BlockGetter;Lnet/minecraft/core/BlockPos;)Z
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftChest.java b/src/main/java/org/bukkit/craftbukkit/block/CraftChest.java
+index c7d19cef79dbe39edc8cddf0c60c7b59d656e581..78e091a4aeb17df728d42930247006e51d26454f 100644
+--- a/src/main/java/org/bukkit/craftbukkit/block/CraftChest.java
++++ b/src/main/java/org/bukkit/craftbukkit/block/CraftChest.java
+@@ -100,4 +100,21 @@ public class CraftChest extends CraftLootable<ChestBlockEntity> implements Chest
+         return getTileEntity().openersCounter.opened;
+     }
+     // Paper end - More Lidded Block API
++
++    // Paper start - More Chest Block API
++    @Override
++    public boolean isBlocked() {
++        if (!isPlaced()) {
++            return false;
++        }
++        if (ChestBlock.isChestBlockedAt(getWorldHandle(), getPosition())) {
++            return true;
++        }
++        if (ChestBlock.getBlockType(this.data) == net.minecraft.world.level.block.DoubleBlockCombiner.BlockType.SINGLE) {
++            return false;
++        }
++        net.minecraft.core.Direction direction = ChestBlock.getConnectedDirection(this.data);
++        return ChestBlock.isChestBlockedAt(getWorldHandle(), getPosition().relative(direction));
++    }
```

Comment:
Add a comment explaining the general area this logic is taken from.

Reply:
Agree, noted.
I've added sanity checks used in source, and the method grew a bit. I'm not using chest's exact check as it's also somewhat hardcoded in a sense that redundant data is fetched & throw-away objects are created, so instead the method mimics it using the same checks & logic. Not sure if that's acceptable, or if something else is preferred.